walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,69 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Set;
+import java.util.List;
+import java.lang.*;
+import java.util.Comparator;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> meetingTimes = new ArrayList<TimeRange>();
+
+    //Request information
+    long requestDuration = request.getDuration();
+    Collection<String> requestAttendees = request.getAttendees();
+
+    // Traverse collection of events to get attendees, and time range of each relevant event
+    for(Event event: events){
+        TimeRange eventRange = event.getWhen();
+
+        // Modify copy of Event Attendees to only include relevant attendees that overlap with the request
+        Set<String> eventAttendees = event.getAttendees();
+        Set<String> copy = new HashSet<>();
```

Comment:
Let's rename this variable to be more descriptive. how about "attendeesCopy" or something similar?

Alternatively we can just have eventAttendees variable be a copy, like so:
`Set<String> eventAttendees = new HashSet<>(event.getAttendees())`